interactor/tests/mvx_esdt_safe_tests.rs

```diff
@@ -0,0 +1,1391 @@
+use common_blackbox_setup::{
+    RegisterTokenArgs, ISSUE_COST, OPERATION_HASH_STATUS_STORAGE_KEY, SOV_TO_MVX_TOKEN_STORAGE_KEY,
+};
+use error_messages::{
+    BANNED_ENDPOINT_NAME, GAS_LIMIT_TOO_HIGH, INVALID_TYPE, NOTHING_TO_TRANSFER,
+    NO_ESDT_SAFE_ADDRESS, PAYMENT_DOES_NOT_COVER_FEE, TOO_MANY_TOKENS,
+};
+use header_verifier::OperationHashStatus;
+use multiversx_sc_snippets::multiversx_sc_scenario::multiversx_chain_vm::crypto_functions::sha256;
+use multiversx_sc_snippets::{hex, imports::*};
+use proxies::fee_market_proxy::{FeeStruct, FeeType};
+use rust_interact::config::Config;
+use rust_interact::mvx_esdt_safe::mvx_esdt_safe_interactor_main::MvxEsdtSafeInteract;
+use rust_interact::{FEE_TOKEN, FIRST_TOKEN, SECOND_TOKEN, SOV_TOKEN, TOKEN_TICKER};
+use serial_test::serial;
+use structs::aliases::PaymentsVec;
+use structs::configs::EsdtSafeConfig;
+use structs::operation::{Operation, OperationData, OperationEsdtPayment, TransferData};
+
+// Test that deposit fails when there is nothing to transfer and fee is disabled
+// Steps:
+// 1. Deploy header verifier smart contract
+// 2. Deploy mvx-esdt-safe smart contract
+// 3. Deploy fee-market smart contract
+// 4. Set fee-market address in mvx-esdt-safe smart contract
+// 5. Unpause mvx-esdt-safe smart contract
+// 6. Deposit tokens
+#[tokio::test]
+#[serial]
+#[cfg_attr(not(feature = "chain-simulator-tests"), ignore)]
+async fn deposit_nothing_to_transfer_no_fee() {
+    let mut chain_interactor = MvxEsdtSafeInteract::new(Config::chain_simulator_config()).await;
+    chain_interactor.deploy_header_verifier().await;
```

Comment:
Please create a function to deploy all these contacts. You can send EsdtConfig, FeeConfig as arguments
- deploy_header_verifier
- deploy_mvx_esdt_safe
- deploy_fee_market
- set_fee_market_address
- unpause_endpoint

Reply:
If you don't need all these deploys in some test, just don't call the function and deploy manually